ts/session/apis/snode_api/swarmPolling.ts

```diff
@@ -414,27 +449,40 @@ export class SwarmPolling {
   public async pollOnceForKey([pubkey, type]: PollForUs | PollForLegacy | PollForGroup) {
     const namespaces = this.getNamespacesToPollFrom(type);
     const swarmSnodes = await SnodePool.getSwarmFor(pubkey);
-    let resultsFromAllNamespaces: RetrieveMessagesResultsBatched | null;
+    let resultsFromAllNamespaces: RetrieveMessagesResultsMergedBatched | null;
 
-    let toPollFrom: Snode | undefined;
+    let toPollFrom: Array<Snode> = [];
 
     try {
-      toPollFrom = sample(swarmSnodes);
+      toPollFrom = sampleSize(swarmSnodes, RETRIEVE_SNODES_COUNT);
 
-      if (!toPollFrom) {
+      if (toPollFrom.length !== RETRIEVE_SNODES_COUNT) {
         throw new Error(
-          `SwarmPolling: pollOnceForKey: no snode in swarm for ${ed25519Str(pubkey)}`
+          `SwarmPolling: pollOnceForKey: not snodes in swarm for ${ed25519Str(pubkey)}. Expected to have at least ${RETRIEVE_SNODES_COUNT}.`
         );
       }
-      // Note: always print something so we know if the polling is hanging
-      window.log.info(
-        `SwarmPolling: about to pollNodeForKey of ${ed25519Str(pubkey)} from snode: ${ed25519Str(toPollFrom.pubkey_ed25519)} namespaces: ${namespaces} `
-      );
-      resultsFromAllNamespaces = await this.pollNodeForKey(toPollFrom, pubkey, namespaces, type);
 
-      // Note: always print something so we know if the polling is hanging
+      const resultsFromAllSnodesSettled = await Promise.allSettled(
+        toPollFrom.map(async snode => {
```

Comment:
map with async is 9.99/10 times always dodgy and doing unexpected things. Should we not use reguarly for loop with async inside?

Reply:
Cool. Resolving